guides/keras_cv/simsiam-with-kerascv.py

```diff
@@ -0,0 +1,604 @@
+
+eval_val_ds = tf.data.Dataset.from_tensor_slices(
+    (x_test, tf.keras.utils.to_categorical(y_test, 10))
+)
+eval_val_ds = eval_val_ds.repeat()
+eval_val_ds = eval_val_ds.shuffle(1024)
+eval_val_ds = eval_val_ds.batch(BATCH_SIZE)
+eval_val_ds = eval_val_ds.prefetch(tf.data.AUTOTUNE)
+
+"""
+## Benchmark Against a Naive Model
+"""
+
+TEST_EPOCHS = 50
+TEST_STEPS_PER_EPOCH = x_raw_train.shape[0] // BATCH_SIZE
+
+
+def get_eval_model(img_size, backbone, total_steps, trainable=True, lr=1.8):
+    backbone.trainable = trainable
+    inputs = tf.keras.layers.Input((img_size, img_size, 3), name="eval_input")
+    x = backbone(inputs, training=trainable)
+    o = tf.keras.layers.Dense(10, activation="softmax")(x)
+    model = tf.keras.Model(inputs, o)
+    cosine_decayed_lr = tf.keras.experimental.CosineDecay(
+        initial_learning_rate=lr, decay_steps=total_steps
+    )
+    opt = tf.keras.optimizers.SGD(cosine_decayed_lr, momentum=0.9)
+    model.compile(optimizer=opt, loss="categorical_crossentropy", metrics=["acc"])
+    return model
+
+
+no_pt_eval_model = get_eval_model(
+    img_size=96,
+    backbone=get_backbone((96, 96, 3)),
+    total_steps=TEST_EPOCHS * TEST_STEPS_PER_EPOCH,
+    trainable=True,
+    lr=1e-3,
+)
+no_pt_history = no_pt_eval_model.fit(
+    eval_train_ds,
+    batch_size=BATCH_SIZE,
+    epochs=TEST_EPOCHS,
+    steps_per_epoch=TEST_STEPS_PER_EPOCH,
+    validation_data=eval_val_ds,
+    validation_steps=VAL_STEPS_PER_EPOCH,
+)
+
+"""
+Pretty bad results!  Lets try fine-tuning our SimSiam pretrained model:
+"""
+
+pt_eval_model = get_eval_model(
+    img_size=96,
+    backbone=contrastive_model.backbone,
+    total_steps=TEST_EPOCHS * TEST_STEPS_PER_EPOCH,
+    trainable=False,
+    lr=30.0,
+)
+pt_eval_model.summary()
+pt_history = pt_eval_model.fit(
+    eval_train_ds,
+    batch_size=BATCH_SIZE,
+    epochs=TEST_EPOCHS,
+    steps_per_epoch=TEST_STEPS_PER_EPOCH,
+    validation_data=eval_val_ds,
+    validation_steps=VAL_STEPS_PER_EPOCH,
+)
+
+"""
+All that is left to do is evaluate the models:
+"""
+print("no pretrain", no_pt_eval_model.evaluate(eval_test_ds))
+print("pretrained", pt_eval_model.evaluate(eval_test_ds))
+"""
+Awesome!  Our pretrained model stomped the non-pretrained model.
+71% accuracy is quite good for a ResNet18 on the STL-10 dataset.
+For better results, try using an EfficientNetV2B0 instead.
+Unfortunately, this will require a higher end graphics card as
+SimSiam has a minimum batch size of 512.
+
+## Conclusion
+
+TensorFlow similarity can be used to easily train KerasCV models using
+contrastive algorithms such as SimCLR, SimSiam and BarlowTwins.
+This allows you to leverage large corpuses of unlabelled data in your
+model trainining pipeline.
+
+Some follow-up exercises to this tutorial:
+
+- Train an EfficientNetV2B0 on STL-10
+- Experiment with other data augmentation techniques in pretraining
+- Train a model using the BarlowTwins implementation in TensorFlow similarity
+"""
```

Comment:
@fchollet we're WAY over the maximum line numbers for keras.io guide.

Simplifying the code should get us to something like 450~

Any other suggestions of what to drop?

Reply:
It doesn't "feel" that long -- there are lots of lines because there's a lot of declarative code (plots, etc). Perhaps it could be refactored a bit for conciseness. Otherwise we can just override the LOC limit.